src/main/java/santiagoAndFerdy/vgs/gridScheduler/GridScheduler.java

```diff
@@ -0,0 +1,98 @@
+package santiagoAndFerdy.vgs.gridScheduler;
+
+import java.net.MalformedURLException;
+import java.rmi.RemoteException;
+import java.rmi.server.UnicastRemoteObject;
+import java.util.HashMap;
+import java.util.Queue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+
+import org.apache.commons.collections4.queue.CircularFifoQueue;
+
+import com.linkedin.parseq.Engine;
+import com.linkedin.parseq.EngineBuilder;
+import com.linkedin.parseq.Task;
+
+import santiagoAndFerdy.vgs.discovery.HeartbeatHandler2;
+import santiagoAndFerdy.vgs.discovery.IHeartbeatSender;
+import santiagoAndFerdy.vgs.discovery.IRepository;
+import santiagoAndFerdy.vgs.model.Request;
+import santiagoAndFerdy.vgs.resourceManager.EagerResourceManager;
+import santiagoAndFerdy.vgs.rmi.RmiServer;
+
+public class GridScheduler extends UnicastRemoteObject implements Runnable {
+    private static final long             serialVersionUID = -5694724140595312739L;
+    private Queue<Request>                jobQueue;
+    private Queue<EagerResourceManager>   idleRM;
+    private HeartbeatHandler2             hHandler;
+    private ScheduledExecutorService      timerScheduler;
+    private Engine                        engine;
+    private HashMap<String, Task<Object>> status;
+    private String                        myURLForHB;
+    private String                        myURL;
+    private IRepository<IHeartbeatSender> repoForHB;
+    private RmiServer                     rmiServer;
+    private boolean                       running;
+    private Thread                        pollingThread;
+
+    public GridScheduler(RmiServer rmiServer, IRepository<IHeartbeatSender> repo, String url) throws RemoteException, MalformedURLException {
+        jobQueue = new LinkedBlockingQueue<>();
+        idleRM = new CircularFifoQueue<>();
+        myURLForHB = url + "-hb";

```

Comment:
shouldn't need a separate url for the heartbeat receiver, should be able to make it so that you can send the heartbeat to the object itself.